src/Menu/MenuManager.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace FluxErp\Menu;
+
+use FluxErp\Models\Permission;
+use Illuminate\Routing\Route;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use Illuminate\Support\Traits\Macroable;
+use Spatie\Permission\Exceptions\PermissionDoesNotExist;
+
+/**
+ * MenuManager Facade
+ *
+ * The MenuManager facade provides an interface to manage and retrieve menu items in FluxErp.
+ * It allows you to register menu items based on Laravel routes and perform various operations on them.
+ *
+ * @method static void register(Route $route, string $label = null, string $icon = null, int $order = null)
+ * @method static void unregister(string $name)
+ * @method static array all()
+ * @method static array|null get(string $name)
+ * @method static array forGuard(string $guard, string $group = null, bool $ignorePermissions = false)
+ *
+ * @see \FluxErp\Menu\MenuManager
+ */
+class MenuManager
+{
+    use Macroable;
+
+    protected array $menuItems = [];
+
+    /**
+     * @throws \Symfony\Component\Routing\Exception\RouteNotFoundException
+     */
+    public function register(
+        Route $route,
+        string $label = null,
+        string $icon = null,
+        int $order = null,
+    ): void {
+        $routeName = $route->getName();
+
+        $guards = array_values(
+            array_filter(
+                $route->middleware(),
+                fn ($guard) => str_starts_with($guard, 'auth:')
+            )
+        );
+        $guard = array_shift($guards);
+        $guard = str_replace('auth:', '', $guard);
+
+        $path = str_contains($routeName, '.') && ! str_ends_with($routeName, '.')
+            ? Str::beforeLast($routeName, '.') . '.children.' . Str::afterLast($routeName, '.')
+            : (str_ends_with($routeName, '.') ? rtrim($routeName, '.') : $routeName);
+
+        data_set($this->menuItems, $path, array_merge(
+            data_get($this->menuItems, $path) ?? [],
+            [
+                'label' => $label ?: Str::afterLast($path, '.'),
+                'uri' => Str::of($route->uri())->start('/')->toString(),
+                'icon' => $icon,
+                'route_name' => $routeName,
+                'guard' => $guard,
+                'domain' => $route->getDomain(),
+                'permission' => $route->getPermissionName(),
+                'order' => $order
+                    ?? count(
```

Comment:
?:

Reply:
If Position is 0 ?: triggers the "Else", ?? Only if null